fix PuTTY link in putty-winscp.mdx

ptgott opened the pull request:
Closes #57966

The link to PuTTY was pointing to putty.org. This domain has no relation to the PuTTY project! Instead, the website run by the actual PuTTY team can be found under https://putty.software , see https://hachyderm.io/@simontatham/115025974777386803

zmb3 reviewed (approved):
Thanks for buddying this, Paul.

ptgott committed:
Fix PuTTY link in putty-winscp.mdx

Use the domain for the PuTTY project, https://putty.software, rather
than the off-topic putty.org domain.
